src/parser/user-extractor-transformer.ts

```diff
@@ -0,0 +1,68 @@
+import configuration from "../configuration/config-reader";
+import { GetActivity } from "../get-activity";
+import { GitHubIssue, GitHubIssueComment } from "../github-types";
+import { Result, Transformer } from "./processor";
+
+/**
+ * Creates entries for each bounty hunter with its associated comments.
+ */
+export class UserExtractorTransformer implements Transformer {
+  private readonly _configuration = configuration["user-extractor"];
+
+  get enabled(): boolean {
+    return true;
+  }
+
+  /**
+   * Checks if the comment is made by a human user, and if it not a command.
+   * @param comment
+   */
+  _checkEntryValidity(comment: GitHubIssueComment) {
+    return !!comment.user && !!comment.body && comment.user.type === "User";
+  }
+
+  _extractBountyPrice(issue: GitHubIssue) {
+    if (this._configuration["redeem-bounty"] === false) {
+      return 0;
+    }
+    const sortedPriceLabels = issue.labels
+      .reduce((acc, label) => {
+        const labelName = typeof label === "string" ? label : label.name;
+        if (labelName?.startsWith("Price: ")) {
+          const price = parseFloat(labelName.replace("Price: ", ""));
+          if (!isNaN(price)) {
+            acc.push(price);
+          }
+        }
+        return acc;
+      }, [] as Array<number>)
+      .sort((a, b) => {
+        return a - b;
+      });
+    if (!sortedPriceLabels.length) {
+      console.warn("There are no price labels in this repository.");
+      return 0;
+    }
+    return sortedPriceLabels[0];
+  }
+
+  transform(data: Readonly<GetActivity>, result: Result): Result {
+    if (data.comments) {
+      for (const comment of data.comments as GitHubIssueComment[]) {
+        if (comment.user && comment.body && this._checkEntryValidity(comment)) {
+          const bounty =
+            (data.self as GitHubIssue)?.assignee?.id === comment.user.id
+              ? {
+                  reward: this._extractBountyPrice(data.self as GitHubIssue),
+                }
+              : undefined;
```

Comment:
```suggestion
              : void 0;
```

I heard this is considered more robust as it is possible for `undefined` to be defined in JavaScript. Not sure if this is too old school though, but thats what I would do in this situation.

Reply:
Except if we plan on supporting IE3, we should be fine
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined#description

Reply:
We only plan to support the latest Chrome browsers basically, or wherever Metamask is commonly used.